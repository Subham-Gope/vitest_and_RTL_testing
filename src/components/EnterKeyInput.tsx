import React, { useState } from "react";
import type { KeyboardEvent } from "react";

interface PropsType {
  message: string;
}

const EnterKeyInput: React.FC<PropsType> = ({ message }) => {
  const [text, setText] = useState<string>("");

  const handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    if (e.key === "Enter") {
      setText("Enter key pressed");
    }
  };

  return (
    <>
      <h1>{message}</h1>
      <div>
        <label htmlFor="enterKeyInput">Enter the Text Here</label>
        <input
          type="text"
          id="enterKeyInput"
          value={text}
          onKeyDown={handleKeyDown}
        ></input>
      </div>
      {text && <p>{text}</p>}
    </>
  );
};

export default EnterKeyInput;
