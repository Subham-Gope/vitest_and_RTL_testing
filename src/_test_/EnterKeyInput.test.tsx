import { describe, test, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import EnterKeyInput from "../components/EnterKeyInput";

describe("Testing the EnterKeyInput.tsx file", () => {
  test("Should detect Enter key press", async () => {
    const user = userEvent.setup();

    const { container } = render(<EnterKeyInput message={"Hello"} />);
    expect(container).toMatchSnapshot();

    const input = screen.getByRole("textbox", { name: /enter the/i });
    expect(input).toBeInTheDocument();

    // simulating the user interaction
    await user.type(input, "Hello{enter}");

    expect(input).toHaveValue("Enter key pressed");

    const text = screen.getByText(/enter key/i);
    expect(text).toBeInTheDocument();
  });
});
