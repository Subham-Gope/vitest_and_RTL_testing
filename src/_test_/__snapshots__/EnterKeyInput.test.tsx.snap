// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the EnterKeyInput.tsx file > Should detect Enter key press 1`] = `
<div>
  <h1>
    Hello
  </h1>
  <div>
    <label
      for="enterKeyInput"
    >
      Enter the Text Here
    </label>
    <input
      id="enterKeyInput"
      type="text"
      value=""
    />
  </div>
</div>
`;
